### PR TITLE
All changes together without offline

### PR DESCRIPTION
**Pull request by cstamas**
All changes from https://github.com/trinodb/trino/pull/20495 but removed "offline".

Changes:
* GH actions: remove "offline" stuff, no need to "prime local repository" either (go-offline and related removed)
* removed takari-local-repository, instead using RRF + file locking + BF collector
* mvnw changed to "script-only"

### CODE REVIEW

**Comment by wendigo** [comment]
@cstamas I wanted to remove go offline for some time as it was causing spurious failures in unexpected scenarios.  Thanks for working on this! We want to stay on the latest Maven to be able to move to 4.0.0 when it's out.

**Comment by cstamas** [comment]
Problem on master is:
* presence of takari-local-repository, if "just removed" fun things happen (and breakage)
* RRF makes some of those issues fixed, other bits fixed are Maven 3.9.x file locking
* fun starts when we enable smart builder across all, but we need some plugin changes, see https://github.com/trinodb/trino-maven-plugin/pull/23

**Comment by cstamas** [comment]
All fine in CI

In the meantime Provisio got PR as well https://github.com/jvanzyl/provisio/pull/65

**Comment by cstamas** [comment]
Now comes the fun part: enable smart builder for all... (trino-maven-plugin and provisio will yell, but IMHO all should be fine... if not, i'll just take a step back).

**Comment by cstamas** [comment]
@wendigo hope you can release proviso as well (I left a comment to be done before merge), and then with both updated with sync to master.

**Comment by cstamas** [comment]
Updated PR, @wendigo pls ping if provisio new release done and merged, Thanks! (or just done, we can updated it as part of this PR even)

**Comment by wendigo** [comment]
@cstamas I've asked @jvanzyl over the Slack to do a release.

**Comment by cstamas** [comment]
Removed central prefixes, basically we are back to "allow all for central" (as I see we do not care for 'leaking reqs' toward Central, which is OK).

**Comment by cstamas** [comment]
Updated PR, now latest trino-maven-plugin and proviso used.

**Comment by wendigo** [comment]
@nineinchnick ptal

File: .mvn/maven.config [hunk 1]
```diff
@@ -1,2 +1,10 @@
 --strict-checksums
-
+-Daether.syncContext.named.factory=file-lock
+-Daether.syncContext.named.nameMapper=file-hgav
+-Daether.syncContext.named.retry=5
```
Comment (nineinchnick) [inline .mvn/maven.config:4]: I can't find any reference docs for these properties, does this mean downloads will be retried?

File: .mvn/maven.config [hunk 2]
```diff
@@ -1,2 +1,10 @@
...
+-Daether.syncContext.named.time=30
```
Comment (nineinchnick) [inline .mvn/maven.config:5]: What's the unit of this?

File: .mvn/maven.config (hunk 1 above)
Comment (cstamas) [reply .mvn/maven.config:4]: https://maven.apache.org/resolver/configuration.html

File: .mvn/maven.config (hunk 2 above)
Comment (cstamas) [reply .mvn/maven.config:5]: aether.syncContext.named.time.unit by default is `TimeUnit.SECONDS`

File: .mvn/maven.config (hunk 1 above)
Comment (nineinchnick) [reply .mvn/maven.config:4]: Thanks, can you add it in a comment in this file? I see `aether.connector.http.retryHandler.count` has a default value of `3`.

File: .mvn/maven.config (hunk 2 above)
Comment (nineinchnick) [reply .mvn/maven.config:5]: `aether.syncContext.named.time` already has a default of `30`, remove it or add the unit too to make it easier to read.

**Review by nineinchnick — APPROVED** [review]
This is great, I'm very happy to remove the `download-maven-dependencies.sh` script. This unblocks some other work. Thanks again!

**Comment by cstamas** [comment]
Last plugin that needs some love:
```
[WARNING] *****************************************************************
[WARNING] The following plugins are not marked as thread-safe in trino-server-rpm:
[WARNING]   io.airlift.maven.plugins:redlinerpm-maven-plugin:2.1.8
[WARNING] 
[WARNING] Enable debug to see precisely which goals are not marked as thread-safe.
[WARNING] *****************************************************************
```

**Comment by cstamas** [comment]
Pushed two more commits:
* "slight" introduction of smart builder (was unused just put in: now where `-T` is used, `-b smart` is used as well)
* apply PR comments: placed doco URL and removed def value

File: .mvn/maven.config (hunk 1 above)
Comment (nineinchnick) [reply .mvn/maven.config:5]: Were these retries always available in Maven?

Comment (cstamas) [reply .mvn/maven.config:5]: Since Maven 3.9.x, with introduction of locking

Comment (cstamas) [reply .mvn/maven.config:5]: In Resolver 2 this page is generated and contains more data, like "since" and "source": https://maven.apache.org/resolver-archives/resolver-2.0.0-alpha-7/configuration.html

BUT, do not use this page for Resolver 1.x (Maven 3.x) as there has been renamed property keys! Still, if you can match the property (unchanged since 1.x or figure out what was renamed from) you can figure out "since".

File: .github/actions/setup/action.yml [hunk 3]
```diff
@@ -3,16 +3,13 @@ description: "Verify checked out commits and setup Java"
 inputs:
   java-version:
     description: "Java version to setup"
-    default: 21
```
Comment (wendigo) [inline .github/actions/setup/action.yml:6]: revert, irrelevant change

File: .github/actions/setup/action.yml [hunk 4]
```diff
@@ -3,16 +3,13 @@ description: "Verify checked out commits and setup Java"
...
+    default: '21'
   cache:
     description: "Cache Maven repo (true/false/restore)"
-    default: true
```
Comment (wendigo) [inline .github/actions/setup/action.yml:9]: revert, irrelevant change

File: .github/actions/setup/action.yml [hunk 5]
```diff
@@ -3,16 +3,13 @@ description: "Verify checked out commits and setup Java"
...
+    default: 'true'
   cleanup-node:
     description: "Clean up node (true/false) to increase free disk space"
-    default: false # Disabled by default as it adds ~4 minutes of test runtime. Should be enabled case by case.
```
Comment (wendigo) [inline .github/actions/setup/action.yml:12]: revert, irrelevant change

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -492,8 +494,9 @@ jobs:
           echo "matrix=$(jq -c '.' matrix.json)" >> $GITHUB_OUTPUT
 
   test:
+    needs:
+      - build-test-matrix
     runs-on: ubuntu-latest
-    needs: build-test-matrix
```
Comment (wendigo) [inline .github/workflows/ci.yml:496]: revert, irrelevant change

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -937,6 +940,8 @@ jobs:
           echo "matrix=$(cat matrix.json)" >> $GITHUB_OUTPUT
 
   pt:
+    needs:
```
Comment (wendigo) [inline .github/workflows/ci.yml:943]: revert, irrelevant change

File: mvnw [hunk 8]
```diff
@@ -21,288 +21,219 @@
 # ----------------------------------------------------------------------------
 # Apache Maven Wrapper startup batch script, version 3.2.0
 #
-# Required ENV vars:
```
Comment (wendigo) [inline mvnw:24]: Why this has changes? It's unrelated, please undo

**Review by wendigo — APPROVED** [review]
LGTM % some unrelated changes to undo

**Comment by wendigo** [comment]
@cstamas please squash commits 1 and 3 together. 2 can be a separate one

File: mvnw (hunk 8 above)
Comment (wendigo) [reply mvnw:24]: Can you extract this to a separate commit?

Comment (wendigo) [reply mvnw:24]: We tend to split changes into smaller, atomic commits (and we test that each commit separately compiles) and keep commit messages in sync with the actual commit content.

**Comment by cstamas** [comment]
Created https://github.com/trinodb/trino/pull/20504 for mvnw

**Comment by wendigo** [comment]
@cstamas mvnw merged

**Comment by wendigo** [comment]
I've amended the commit message (to reword and drop mvn wrapper mention as it was extracted to a separate commit) and pushed directly to master as https://github.com/trinodb/trino/commit/f133a42e4e1df7c7dc8777edf45df71b3fa0d1b6. Thank you @cstamas !

File: .github/workflows/ci.yml [hunk 9]
```diff
@@ -97,7 +97,8 @@ jobs:
                 format('refs/pull/{0}/head', github.event.client_payload.pull_request.number) || '' }}
       - uses: ./.github/actions/setup
         with:
-          cleanup-node: true
+          cache: 'restore'
```
Comment (nineinchnick) [inline .github/workflows/ci.yml:100]: Looks like no job populates the cache anymore.

Comment (nineinchnick) [inline .github/workflows/ci.yml:100]: Looks like no job populates the cache anymore, all uses of `./.github/actions/setup` are done with `cache: restore`.

File: .github/workflows/ci.yml [hunk 10]
```diff
@@ -18,14 +18,13 @@ env:
   # An envar that signals to tests we are executing in the CI environment
   CONTINUOUS_INTEGRATION: true
   # allow overriding Maven command
-  MAVEN: ./mvnw --offline
-  # maven.wagon.rto is in millis, defaults to 30m
-  MAVEN_OPTS: "-Xmx512M -XX:+ExitOnOutOfMemoryError -Dmaven.wagon.rto=60000"
```
Comment (findepi) [inline .github/workflows/ci.yml:23]: Why `-Xmx512M -XX:+ExitOnOutOfMemoryError` was removed?
is this related to removal of the go-offline?

File: .github/workflows/ci.yml [hunk 11]
```diff
@@ -18,14 +18,13 @@ env:
...
-  MAVEN_FAST_INSTALL: "-B --strict-checksums -V --quiet -T 1C -DskipTests -Dmaven.source.skip=true -Dair.check.skip-all"
-  MAVEN_COMPILE_COMMITS: "-B --strict-checksums --quiet -T 1C -DskipTests -Dmaven.source.skip=true -Dair.check.skip-all=true -Dmaven.javadoc.skip=true --no-snapshot-updates --no-transfer-progress -pl '!:trino-server-rpm'"
+  MAVEN: ./mvnw
+  MAVEN_OPTS: ""
+  MAVEN_INSTALL_OPTS: ""
+  MAVEN_FAST_INSTALL: "-B -V --quiet -T 1C -b smart -DskipTests -Dmaven.source.skip=true -Dair.check.skip-all"
+  MAVEN_COMPILE_COMMITS: "-B --quiet -T 1C -b smart -DskipTests -Dmaven.source.skip=true -Dair.check.skip-all=true -Dmaven.javadoc.skip=true --no-snapshot-updates --no-transfer-progress -pl '!:trino-server-rpm'"
   MAVEN_GIB: "-P gib -Dgib.referenceBranch=refs/remotes/origin/${{ github.event_name == 'pull_request' && github.event.pull_request.base.ref || github.event.repository.default_branch }}"
-  MAVEN_TEST: "-B --strict-checksums -Dmaven.source.skip=true -Dair.check.skip-all --fail-at-end -P gib -Dgib.referenceBranch=refs/remotes/origin/${{ github.event_name == 'pull_request' && github.event.pull_request.base.ref || github.event.repository.default_branch }}"
```
Comment (findepi) [inline .github/workflows/ci.yml:28]: is `--strict-checksums` no longer necessary?

File: .github/workflows/ci.yml (hunk 10 above)
Comment (findepi) [reply .github/workflows/ci.yml:23]: https://github.com/trinodb/trino/pull/21256

File: .github/workflows/ci.yml (hunk 11 above)
Comment (findepi) [reply .github/workflows/ci.yml:28]: https://github.com/trinodb/trino/pull/21256
